plugins/rebalance/rest.go

```diff
@@ -0,0 +1,247 @@
+	txn.Ctx.Logger().WithField("volname", rebal.Volname).Info("rebalance started")
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, rebalance)
+}
+
+func rebalanceStop(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	// Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	rebalinfo, err := GetRebalanceInfo(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check rebalance process is started or not
+	if rebalinfo.Status != rebalanceapi.GfDefragStatusStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance process is not started on this volume",
+			api.ErrCodeDefault)
+		return
+	}
+
+	// Check remove brick operation is running
+	//TODO
+
+	//A simple transaction to stop rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "rebal-stop.Commit",
+			Nodes:  txn.Nodes,
+		},
+		{
+			DoFunc: "rebal-stop.StoreVolume",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set request in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	rebalinfo.Volname = volname
+	rebalinfo.Status = rebalanceapi.GfDefragStatusStopped
+	rebalinfo.Cmd = rebalanceapi.GfDefragCmdStop
+
+	err = txn.Ctx.Set("rinfo", rebalinfo)
+	if err != nil {
+		logger.WithError(err).Error("failed to set rebalance info in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to stop rebalance on volume")
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+	status := getStatus(rebalinfo)
+	rebalance := fmt.Sprintf("Status:%+v ,volume rebalance:%s: success: rebalance process may be in the middle of a file migration.The process will be fully stopped once the migration of the file is complete. Please check rebalance process for completion before doing any further brick related tasks on the volume.", status, rebalinfo.Volname)
```

Comment:
Printing everything in API output is not Good. API response should be JSON so that CLI or any other consumers can easily parse the output. Send rebalinfo as response.